example_js/ggml_chat.js

```diff
@@ -0,0 +1,57 @@
+import { GGMLChatCompletionRequest, GGMLChatPrompt } from '_wasi_nn_ggml_template'
+import { build_graph_from_cache } from '_wasi_nn_ggml'
+function main() {
+    let opt = {
+        "enable-log": true,
+        "ctx_size": 512,
+        "n-predict": 1024,
+        "n-gpu-layers": 100,
+        "batch-size": 512,
+        "temp": 0.8,
+        "repeat-penalty": 1.1
+    }
+
+    let graph = build_graph_from_cache(3, JSON.stringify(opt), "default")
+    let context = graph.init_execution_context()
+
+    let template = new GGMLChatPrompt('llama-2-chat')
+
+    let req = new GGMLChatCompletionRequest()
+
+    let messages = ['hello', 'who are you?']
+
+    for (var i in messages) {
+        print("[YOU]:", messages[i])
+        req.push_message("user", messages[i])
+        let p = template.build(req)
+        context.set_input(0, p, [1], 3)
+        var ss = '';
+
+        while (1) {
+            try {
+                context.compute_single()
```

Comment:
Can we use streaming responses? In fact, we could have two examples. One uses the simple response and the other uses streaming responses.